zebra/zebra_rib.c

```diff
@@ -3728,6 +3728,7 @@ static int handle_pw_result(struct zebra_dplane_ctx *ctx)
 	}
 
 done:
+	dplane_ctx_fini(&ctx);
```

Comment:
Wouldn't it be better for the pattern of calling dplane_ctx_fini be handled right after the switch statement in rib_process_dplane_results?  This way when we keep adding new things for the dplane to do, we won't keep having this memory leak?

Reply:
that'd work fine in this specific case, but to do it for every type of update I'd have to look at all the others to see if anyone keeps the ctx around for anything - like as a thread/event argument. I agree it's not great for me to have left this tempting opportunity for leaks lying around.

Reply:
If we keep the context around then we need to have a warning at the top of the switch statement to tell people that this is the pattern to look for.  I spent about 20-25 minutes looking at the code and did not pick up on this pattern at all.  This is dangerous and should be rectified

Reply:
ok, I hear you, I will fix it. would you like to take this one-off for now, or have me fold this in with the more general change?

Reply:
I'll get this one in so we can stop the memory leak.  This is not good for a system that has interface going up / down w/ pseudo wires.  We'll leak memory pretty badly